lib/elixir/lib/range.ex

```diff
@@ -40,7 +40,8 @@ defmodule Range do
   defstruct first: nil, last: nil
 
   @type t :: %Range{}
-  @type t(first, last) :: %Range{first: first, last: last}
+  @type t(first :: integer, last :: integer) :: %Range{first: integer, last: integer}
```

Comment:
I think we should still pass first and last to the second part because someone may want to restrict it to a non_neg_integer or something of sorts.